movieapp/src/services/utils.ts

```diff
@@ -1,9 +1,32 @@
+import queryString, { ParsedQuery } from 'query-string';
+
 type SearchParams = {
   searchBy: string;
-  filter: string;
+  genre: string;
   sortBy: string;
+  keyword: string;
+  movie: string;
 };
 
-export function searchParamsToQueryString(params: SearchParams | null): string {
-  return `http://localhost:4000/movies?searchBy=${params?.searchBy || ''}&filter=${params?.filter || ''}&sortBy=${params?.sortBy || 'id'}&sortOrder=desc`;
+export function updateBrowserUrl(params: SearchParams | queryString.ParsedQuery): string {
+  return queryString.stringify({ ...queryString.parseUrl(window.location.href).query, ...params });
+}
+
+export function searchParamsToQueryString(params: SearchParams | ParsedQuery): string {
```

Comment:
again - it's not util, because it's not common 

searchParamsToQueryString should get object and create its key-values into the string:

`?user=Aliakei&sex=male&age=18`